test/test_common.c

```diff
@@ -219,44 +273,65 @@ sport_new (const char *optarg, struct prog *prog)
     sport->packs_in = NULL;
     sport->fd = -1;
     char *const addr = strdup(optarg);
+    char ip[INET6_ADDRSTRLEN];
 #if __linux__
     char *if_name;
     if_name = strrchr(addr, ',');
     if (if_name)
     {
         strncpy(sport->if_name, if_name + 1, sizeof(sport->if_name) - 1);
-        sport->if_name[ sizeof(sport->if_name) - 1 ] = '\0';
+        sport->if_name[sizeof(sport->if_name) - 1] = '\0';
         *if_name = '\0';
     }
     else
         sport->if_name[0] = '\0';
 #endif
     char *port = strrchr(addr, ':');
-    if (!port)
-        goto err;
-    *port = '\0';
-    ++port;
-    if ((uintptr_t) port - (uintptr_t) addr > sizeof(sport->host))
-        goto err;
-    memcpy(sport->host, addr, port - addr);
+    if (!port)/*IpAdress wasn't specified by the user*/
+    {
+        if (get_Ip_from_DNS(prog->prog_hostname, ip, ipv6, addr, INET6_ADDRSTRLEN) == 1)
+        {
+            LSQ_ERROR("Couldn't resolve the hostname.\n");
+            goto err;
+        }
+        port = strdup(optarg);
+
+        memcpy(sport->host, ip, sizeof(ip));
+    }
+    else /*The user specified an ipaddress*/
+    {
+        *port = '\0';
+        ++port;
+        if ((uintptr_t)port - (uintptr_t)addr > sizeof(sport->host))
+            goto err;
+        memcpy(sport->host, addr, port - addr);
+
+        strcpy(ip, addr); /*copy addr to ip so we only need to have the inet_pton code once*/
+    }
+
+    struct sockaddr_in  *const sa4 = (void *)&sport->sas;
+    struct sockaddr_in6 *const sa6 = (void *)&sport->sas;
 
-    struct sockaddr_in  *const sa4 = (void *) &sport->sas;
-    struct sockaddr_in6 *const sa6 = (void *) &sport->sas;
-    if        (inet_pton(AF_INET,  addr, &sa4->sin_addr)) {
+    if (inet_pton(AF_INET, ip, &sa4->sin_addr))
+    {
         sa4->sin_family = AF_INET;
-        sa4->sin_port   = htons(atoi(port));
-    } else if (memset(sa6, 0, sizeof(*sa6)),
-                    inet_pton(AF_INET6, addr, &sa6->sin6_addr)) {
+        sa4->sin_port = htons(atoi(port));
+    }
+    else if (memset(sa6, 0, sizeof(*sa6)), inet_pton(AF_INET6, ip, &sa6->sin6_addr))
+    {
         sa6->sin6_family = AF_INET6;
-        sa6->sin6_port   = htons(atoi(port));
-    } else
+        sa6->sin6_port = htons(atoi(port));
+    }
+    else
+    {
         goto err;
-
+    }
+    
     free(addr);
     sport->sp_prog = prog;
     return sport;
 
-  err:
+err:
```

Comment:
Whitespace-only.

Reply:
This one was probably due to copy pasting code.  The other whitespace changes I don't remember making. Especially since some are in functions I didn't work with. I assume this were changes Visual Stuido made automatically probably when I changed all tabs to spaces.